### ISSUE TITLE
Not sure this is a bug or feature but entire video's preview thumbnails get generated irrespective of video playback position on seekbar. Might consume more data unintentionally.

### ISSUE THREAD

**Issue by desmonk** [TeamNewPipe/NewPipe#9576]
### Checklist

- [X] I am able to reproduce the bug with the [latest version](https://github.com/TeamNewPipe/NewPipe/releases/latest).
- [X] I made sure that there are *no existing issues* - [open](https://github.com/TeamNewPipe/NewPipe/issues) or [closed](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I have read the [FAQ](https://newpipe.net/FAQ/) and my problem isn't listed.
- [X] I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.
- [X] This issue contains only one bug.
- [X] I have read and understood the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md).

### Affected version

0.24.1

### Steps to reproduce the bug

1. Play a video to say few minutes and hit pause.
2. Scroll across seek bar and see how preview thumbnails behave.
3. You will see that entire video's preview thumbnails get loaded.

### Expected behavior

Preview thumbnails should not get loaded beyond user selection otherwise this can result in more data consumption along with more memory and storage space usage.

### Actual behavior

Whole video thumbnails get loaded regardless.

### Affected Android/Custom ROM version

Android 10+

### Affected device model

Many

**Comment by desmonk** [comment]
That option would turn off this feature completely and that is not a solution. App should not load all preview thumbnails in advanced unless user has chosen to do so. This is an issue otherwise resulting more data consumption.

**Comment by ShareASmile** [comment]
Yes, that's a fair point,

> App should not load entire video's preview thumbnails in advance unless user has chosen to do so by using seek bar navigation.

I would request reopening of the issue @SameenAhnaf

**Comment by ShareASmile** [comment]
> Please edit the title to Stop Loading Preview Thumbnails if "Don't Show" is selected in Settings

Yes agree with @SameenAhnaf  issue Title should be precise to feature request @desmonk would you rename issue Title as described above or something similar.

btw Thank You for reopening the issue SameenAhnaf.

**Comment by desmonk** [comment]
@SameenAhnaf, I will clarify as I see how this gets confusing. Preview thumbnails should only get generated upon user selection. 

Meaning if I want to preview till say 01:01:01 time then only up until that, previews should get loaded and not a second more than that as that would result in data consumption without intention.

Option to stop will end loading completely. And this is not what is this post refers to.

**Comment by SC1040-TS2** [comment]
Would it not be more CPU or memory-intensive and inefficient to have to fetch and cache the seekbar thumbnail preview frames every single time the user moves the seekbar position, such as beyond a specified video time duration to load and cache them into memory as @desmonk described for a mitigation of this sort of functional inefficiency?

Additionally, how would the specified time duration to cache preview frames work as a setting, if not just having a user specify a set amount of minutes or whatever timing it desires to cache all seekbar thumbnail preview frames into memory before having to resort to the same inefficient new loading of them into and out of memory after that point is reached in a long enough video?
